### PR TITLE
Implement self interest

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Part of v15 upgrade - expose and utilise self-interest constraints.

#### Primary reviewers
@jacquese

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h [hunk 1]
```diff
@@ -70,12 +68,10 @@ class SPATIALGDK_API InterestFactory
 
 	// Defined Constraint AND Level Constraint
 	void AddPlayerControllerActorInterest(Interest& OutInterest, const AActor* InActor, const FClassInfo& InInfo) const;
-	// Self interests require the entity ID to know which entity is "self". This would no longer be required if there was a first class self
-	// constraint. The components clients need to see on entities they are have authority over that they don't already see through
-	// authority.
-	void AddClientSelfInterest(Interest& OutInterest, const Worker_EntityId& EntityId) const;
+	// The components clients need to see on entities they are have authority over that they don't already see through authority.
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h:71]: I realise this was me but
```suggestion
	// The components clients need to see on entities they have authority over that they don't already see through authority.
```

File: SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h [hunk 2]
```diff
@@ -54,7 +54,7 @@ struct QueryConstraint
 	TSchemaOption<uint32> ComponentConstraint;
 	TArray<QueryConstraint> AndConstraint;
 	TArray<QueryConstraint> OrConstraint;
-	// TSchemaOption<> SelfConstraint; TODO
+	TSchemaOption<bool> SelfConstraint;
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h:57]: why an optional bool? Could it be a sort of void optional, or just a bool instead? I'm not sure what it means for this to be set but false.

**Review by jacquese — APPROVED** [review]
AMAZING thanks for plumbing it through!

File: SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h (hunk 2 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h:57]: Setting it to true or false would set it, but I see that's confusing. Can't have a option<void> so will just make it a bool.

**Review by improbable-valentyn — APPROVED** [review]

**Pull request closed without merging**